core/trino-main/src/main/java/io/trino/operator/OperatorContext.java

```diff
@@ -532,6 +534,16 @@ public static Metrics getOperatorMetrics(Metrics operatorMetrics, long inputPosi
         return operatorMetrics.mergeWith(new Metrics(ImmutableMap.of("Input distribution", new TDigestHistogram(digest))));
     }
 
+    public static Metrics getConnectorMetrics(Metrics connectorMetrics, long physicalInputReadTimeNanos)
+    {
+        if (physicalInputReadTimeNanos == 0) {
```

Comment:
I am not a big fan of using 0 as a special marker. Can we use OptionalLong. Or if maybe -1 so we use value which is out of domain for valid values?

Reply:
I will leave it as is. It's not a maker really, but rather skipping meaningless stat. 
We do something similar for other things, e.g:
https://github.com/trinodb/trino/commit/39f48d5c1423729a0c17445d12271b2cd6dfbdf1
```
            if (node.isMaySkipOutputDuplicates()) {
                nodeOutput.appendDetailsLine("maySkipOutputDuplicates = %s", node.isMaySkipOutputDuplicates());
            }
```
If I used `-1` then I would have to take care of it while doing addition (synchronization). Similaraly `Optional` would require some synchronization and locking.